COMPATIBILITY.md

```diff
@@ -0,0 +1,5 @@
+Server Version: 5.0
+- BSON field 'listCollections.collation' is an unknown field.
+
```

Comment:
Note: this is a new documentation file required by the Continuous Matrix Testing design. It lists any server compatibility issues this version of the driver has, by server version. The corresponding failing tests should be skipped after they've been documented here, so that the continuous testing matrix can be green.